text/0000-binary-ops-specialization.md

```diff
@@ -45,88 +49,102 @@ where
+and lay down some discipline.
 
 # Guide-level explanation
 [guide-level-explanation]: #guide-level-explanation
 
-## New operator traits
-[new-operator-traits]: #new-operator-traits
+## Default operator traits
+[default-operator-traits]: #default-operator-traits
+
+This proposal adds secondary overload traits for all overloadable binary
+operators where the right-hand operand type is generic.
+The traits are named `DefaultPartialEq`, `DefaultPartialOrd`, `DefaultAdd`, etc.
+and defined with the same type parameters and method signatures as the plain
+old Rust 1.0 operator overload traits.
 
-This proposal adds second-generation traits for all binary operators defined in
-the standard library where the right-hand operand type is defined generically.
-The traits are named `PartialEq2`, `PartialOrd2`, `Add2`, etc. and defined
-with the same method signatures as their Rust 1.0 counterparts. Example
-for `PartialEq2`:
+Example for `DefaultPartialEq`:
 
 ```rust
-pub trait PartialEq2<Rhs: ?Sized = Self> {
-    fn eq2(&self, other: &Rhs) -> bool;
-    fn ne2(&self, other: &Rhs) -> bool { !self.eq2(other) }
+pub trait DefaultPartialEq<Rhs: ?Sized = Self> {
+    fn eq(&self, other: &Rhs) -> bool;
+    fn ne(&self, other: &Rhs) -> bool {
+        !self.eq(other)
+    }
 }
 ```
 
-## Default blanket implementation rule
-[default-blanket-implementation-rule]: #default-blanket-implementation-rule
+## Default implementation rule
+[default-implementation-rule]: #default-implementation-rule
 
 Types that need to work as an operand type in binary operators broadly
-fall into two categories. One category is use case specific representations
-of an underlying data type that usually provides binary operators on itself.
-For example, `String` is the owned version of `str`, and `PathBuf` is that
-for `Path`. The underlying types themselves are counted in for the purposes
-of the following rule.
-These types usually implement `Borrow` to the underlying type, and their
-Rust 1.0 binary operator trait impls tend to cover any pairs with other types
-that satisfy the same parameterized `Borrow`. The rule for a crate defining
-such a type is to also define default impls of the new operator traits
-described in this RFC, where a generic type parameter bound by `Borrow`
-defines the operand type other than `Self`:
+fall into two categories. One category is different purpose-specific
+representations of an underlying data type that provides binary
+operators acting on itself. We'll call it the **base operand type** for the
+purposes of this proposal.
+For example, `String` is the standard owned counterpart of `str`,
+and `PathBuf` is this for `Path`. The base operand type itself is
+considered together with its operand type family for the following rule.
+Types in each such family usually implement `Borrow` to the base operand type,
+and their binary operator trait impls, as currently provided, tend to cover
+any possible pairs with the other types in the family.
```

Comment:
Why not just impl `PartialEq` and such for just the base varaints? Then force people to convert the owned versions to do the comparison? Is that too much of an ergonomic hit?

Reply:
Yes, using `.as_str()` and `.as_bytes()` everywhere for a workalike type with a different ownership model is incredibly annoying and inelegant. Especially considering that we have other generic APIs using `Borrow` where all borrower types for the same target work interchangeably.